client/restarts_test.go

```diff
@@ -0,0 +1,39 @@
+package client
+
+import (
+	"github.com/hashicorp/nomad/nomad/structs"
+	"testing"
+	"time"
+)
+
+func TestTaskRunner_ServiceRestartCounter(t *testing.T) {
+	rt := newRestartTracker(structs.JobTypeService, &structs.RestartPolicy{Attempts: 2, Interval: 2 * time.Minute, Delay: 1 * time.Second})
+	rt.increment()
```

Comment:
You should test that within the allowed attempts the return value from nextRestart == delay and that after it is till after the window